Context manager for constants

pllim opened the pull request:
Fix #6950

pllim commented on astropy/constants/utils.py:
```diff
@@ -0,0 +1,35 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Utility functions for ``constants`` sub-package."""
```
@adrn , @eteq thinks you might appreciate this.

mhvk commented:
Overall, this looks nice! One question, maybe not do-able at all, but would it be possible to do `with const.astropyconst13`, i.e., at some level have each constants sub-module be a context? (I hasten to add that I only partially understand contexts, but perhaps one can define `__enter__` and `__exit__` even on a module?)

mhvk commented:
On the name, I ended up thinking we should just have a clear name, not use the modules. How about `set_enabled_constants`? This in analogy with `u.set_enabled_units`. Eventually, we might then also have `add_enabled_constants`.

Indeed, is there room for re-using some of the units constants machinery?

pllim commented:
Ops. I made a mental note to add docstring and must have forgotten about it!

@mhvk -- That sounds like an elegant idea but won't make it to feature freeze. If you think it is worth pursuing instead of this one, perhaps we can re-milestone for 3.1 unless you would like to take a stab at it by Friday.

Thank you for all your comments!

mhvk commented:
@pllim - how about just calling it `set_enabled_constants` - we can then extend for 3.1, but won't have to change the name. That version in units works the same as what you have: nothing is available except what you explicitly allow.

pllim commented:
@stargaser and @mhvk , I think I have addressed the comments thus far. I agree that improvements could be made as a follow-up of this PR and has added a note to this effect in the code.

As for only supporting `astropyconst13` in the PR, I thought I would start small (the minimal amount of support that is sufficient) and see how things go. If there is enough demand to justify adding more, we can always expand in the future. I am afraid to support arbitrary combo now (CODATA2010 + IAU2015) would overly complicate things, and I am not even sure if anyone would use such a combo. So, that is my thought process for the record.

stargaser commented:
I am satisfied with these changes, and the plan for following up in version 3.1.

pllim commented:
@mhvk , do you have any objections? If not, please approve and merge. Thanks!
